### PR TITLE
Remove $elemMatch and $slice projection operators

### LINKED ISSUE #1691: Remove `$elemMatch` and `$slice` projection operators
We should remove support for `$elemMatch` and `$slice` projection operators: they have problems, and we could not fix them before 1.0, given that we still fix issues with comparison.

We should remove only those _projection_ operators and associated code. We should keep the `$elemMatch` _query_ operator. We should also keep support for `find`'s `projection` argument (that would be only a list of fields in our case).

DoD:
- remove tests
- remove code
- update user documentation for supported opeartors

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #1691.

## Readiness checklist

* [ ] I added unit tests for new functionality or bugfixes.
* [ ] I added integration tests for new functionality or bugfixes.
* [ ] I added compatibility tests for new functionality or bugfixes.
* [ ] I made spot refactoring.
* [x] I updated the documentation.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by noisersup — DISMISSED** [review]

File: internal/handlers/common/projection.go [hunk 1]
```diff
@@ -38,20 +34,8 @@ func isProjectionInclusion(projection *types.Document) (inclusion bool, err erro
...
-
-				switch projectionType {
-				case "$elemMatch":
-					inclusion = true
-				case "$slice":
-					inclusion = false
-				default:
-					panic(projectionType + " not supported")
-				}
+				err = lazyerrors.Errorf("projection of %s is not supported", projectionType)
```
Comment (rumyantseva) [inline internal/handlers/common/projection.go:37]: I see this `lazyerror` was already in the code before. I'm not sure why we don't return `NewCommandErrorMsgWithArgument(ErrNotImplemented, ...))` in this case though.

File: internal/handlers/common/projection.go [hunk 2]
```diff
@@ -168,286 +152,10 @@ func projectDocument(inclusion bool, doc *types.Document, projection *types.Docu
...
-				doc.Set(k1, types.Null)
-				return nil
-			}
-
-			doc.Set(k1, res)
 		default:
-			return NewCommandError(ErrCommandNotFound,
-				lazyerrors.Errorf("applyComplexProjection: unknown projection operator: %q", projectionType),
-			)
+			return lazyerrors.Errorf("projection of %s is not supported", projectionType)
```
Comment (rumyantseva) [inline internal/handlers/common/projection.go:156]: Do we need the whole switch at all if we don't put any logic?

Or alternatively, something like this (so we keep the old error as is and use "not implemented" error for the parts that are not implemented:
```go
switch projectionType {
		case "$elemMatch", "$slice":
                      return NewCommandError(ErrNotImplemented, ...))
                default:
                      return NewCommandError(ErrCommandNotFound, ...))
}
```

**Review by rumyantseva — COMMENTED** [review]
A couple of minor comments about the errors we return.

File: internal/handlers/common/projection.go (hunk 2 above)
Comment (chilagrow) [reply internal/handlers/common/projection.go:156]: Nice suggestion thanks :+1:

File: website/docs/reference/supported_commands.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ sidebar_position: 1
 | `find`            |                         | ✅     | Basic command is fully supported                          |
 |                   | `filter`                | ✅     |                                                           |
 |                   | `sort`                  | ✅     |                                                           |
-|                   | `projection`            | ✅     |                                                           |
+|                   | `projection`            | ✅     | Basic projection with field is supported                  |
```
Comment (AlekSi) [inline website/docs/reference/supported_commands.md:24]:
```suggestion
|                   | `projection`            | ✅     | Basic projections with fields is supported                  |
```

Probably? I'm not sure

File: website/docs/reference/supported_commands.md [hunk 4]
```diff
@@ -108,6 +108,17 @@ The following operators and modifiers are available in the `update` and `findAnd
...
+### Projection Operators
+
+The following operators are available in the `find` command `projection` argument.
+
+| Operator     | Status | Comments       |
+|--------------|--------|----------------|
+| `$`          | ❌      | Unimplemented  |
+| `$elemMatch` | ❌      | Unimplemented  |
+| `$meta`      | ❌      | Unimplemented  |
+| `$slice`     | ❌      | Unimplemented  |
```
Comment (AlekSi) [inline website/docs/reference/supported_commands.md:120]: Let's add issues links (and create them if needed)

**Review by AlekSi — COMMENTED** [review]
(I reviewed only docs)

File: website/docs/reference/supported_commands.md (hunk 3 above)
Comment (chilagrow) [reply website/docs/reference/supported_commands.md]: That sounds better than what I wrote, so lets take that :hugs:

File: website/docs/reference/supported_commands.md (hunk 4 above)
Comment (chilagrow) [reply website/docs/reference/supported_commands.md]: Added :+1:

**Review by Fashander — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**